Feature/home-search-screen

guayJheng opened the pull request.

FC4RICA committed:
Refactor: Add nested navigation for Feed

- Created `SearchScreen` and `ServiceDetailScreen` in the `feed` sub-package.
- Added nested navigation for `SearchScreen` and `ServiceDetailScreen` within the Feed section.
- Modified `Screen` to define nested screen routes.
- Updated `HomeScreen` to include the bottom bar only for bottom navigation routes.
- Updated `FeedScreen` to have a button that navigates to `ServiceDetailScreen`.

FC4RICA committed:
Refactor: Add nested navigation for Feed

- Created `SearchScreen` and `ServiceDetailScreen` in the `feed` sub-package.
- Added nested navigation for `SearchScreen` and `ServiceDetailScreen` within the Feed section.
- Modified `Screen` to define nested screen routes.
- Updated `HomeScreen` to include the bottom bar only for bottom navigation routes.
- Updated `FeedScreen` to have a button that navigates to `ServiceDetailScreen`.

FC4RICA committed:
Refactor: Add nested navigation for Feed

- Created `SearchScreen` and `ServiceDetailScreen` in the `feed` sub-package.
- Added nested navigation for `SearchScreen` and `ServiceDetailScreen` within the Feed section.
- Modified `Screen` to define nested screen routes.
- Updated `HomeScreen` to include the bottom bar only for bottom navigation routes.
- Updated `FeedScreen` to have a button that navigates to `ServiceDetailScreen`.

FC4RICA committed:
Refactor: Add nested navigation for Feed

- Created `SearchScreen` and `ServiceDetailScreen` in the `feed` sub-package.
- Added nested navigation for `SearchScreen` and `ServiceDetailScreen` within the Feed section.
- Modified `Screen` to define nested screen routes.
- Updated `HomeScreen` to include the bottom bar only for bottom navigation routes.
- Updated `FeedScreen` to have a button that navigates to `ServiceDetailScreen`.

FC4RICA committed:
Refactor: Add nested navigation for Feed

- Created `SearchScreen` and `ServiceDetailScreen` in the `feed` sub-package.
- Added nested navigation for `SearchScreen` and `ServiceDetailScreen` within the Feed section.
- Modified `Screen` to define nested screen routes.
- Updated `HomeScreen` to include the bottom bar only for bottom navigation routes.
- Updated `FeedScreen` to have a button that navigates to `ServiceDetailScreen`.

FC4RICA committed:
Feat: Implement nested profile screens and navigation

- Added `OrdersStatusScreen`, `OrderStatusDetailScreen`, `PreviousVowsScreen`, `PreviousVowDetailScreen`, and `ProfileSettingScreen` for the profile section.
- Defined new navigation routes for these screens in `Screen.kt`.
- Implemented navigation to these nested profile screens within the `Home` composable, including handling dynamic parameters.
- Added new screens' composables in relevant package.

FC4RICA committed:
Feat: Implement nested profile screens and navigation

- Added `OrdersStatusScreen`, `OrderStatusDetailScreen`, `PreviousVowsScreen`, `PreviousVowDetailScreen`, and `ProfileSettingScreen` for the profile section.
- Defined new navigation routes for these screens in `Screen.kt`.
- Implemented navigation to these nested profile screens within the `Home` composable, including handling dynamic parameters.
- Added new screens' composables in relevant package.

FC4RICA committed:
Feat: Implement nested profile screens and navigation

- Added `OrdersStatusScreen`, `OrderStatusDetailScreen`, `PreviousVowsScreen`, `PreviousVowDetailScreen`, and `ProfileSettingScreen` for the profile section.
- Defined new navigation routes for these screens in `Screen.kt`.
- Implemented navigation to these nested profile screens within the `Home` composable, including handling dynamic parameters.
- Added new screens' composables in relevant package.

FC4RICA committed:
Feat: Implement nested profile screens and navigation

- Added `OrdersStatusScreen`, `OrderStatusDetailScreen`, `PreviousVowsScreen`, `PreviousVowDetailScreen`, and `ProfileSettingScreen` for the profile section.
- Defined new navigation routes for these screens in `Screen.kt`.
- Implemented navigation to these nested profile screens within the `Home` composable, including handling dynamic parameters.
- Added new screens' composables in relevant package.

FC4RICA committed:
Feat: Implement nested profile screens and navigation

- Added `OrdersStatusScreen`, `OrderStatusDetailScreen`, `PreviousVowsScreen`, `PreviousVowDetailScreen`, and `ProfileSettingScreen` for the profile section.
- Defined new navigation routes for these screens in `Screen.kt`.
- Implemented navigation to these nested profile screens within the `Home` composable, including handling dynamic parameters.
- Added new screens' composables in relevant package.

FC4RICA committed:
Feat: Add MyReviews screen and navigation

- Created a new `MyReviewsScreen` under the `profile` package.
- Added `MyReviews` to the `Screen` sealed class for navigation.
- Added a new route for `MyReviewsScreen` in the navigation graph.

FC4RICA committed:
Feat: Add MyReviews screen and navigation

- Created a new `MyReviewsScreen` under the `profile` package.
- Added `MyReviews` to the `Screen` sealed class for navigation.
- Added a new route for `MyReviewsScreen` in the navigation graph.

FC4RICA committed:
Feat: Add MyReviews screen and navigation

- Created a new `MyReviewsScreen` under the `profile` package.
- Added `MyReviews` to the `Screen` sealed class for navigation.
- Added a new route for `MyReviewsScreen` in the navigation graph.

FC4RICA committed:
Feat: Add MyReviews screen and navigation

- Created a new `MyReviewsScreen` under the `profile` package.
- Added `MyReviews` to the `Screen` sealed class for navigation.
- Added a new route for `MyReviewsScreen` in the navigation graph.

FC4RICA committed:
Feat: Add MyReviews screen and navigation

- Created a new `MyReviewsScreen` under the `profile` package.
- Added `MyReviews` to the `Screen` sealed class for navigation.
- Added a new route for `MyReviewsScreen` in the navigation graph.

FC4RICA committed:
Refactor: Implement static color scheme and update home screen UI

FC4RICA committed:
Refactor: Implement static color scheme and update home screen UI

FC4RICA committed:
Refactor: Implement static color scheme and update home screen UI

FC4RICA committed:
Refactor: Implement static color scheme and update home screen UI

FC4RICA committed:
Refactor: Update navigation bar style and icons, and modify secondary color

- Changed the navigation bar's style by setting its container color to primary, content color to onPrimary, and adding rounded corners.
- Added a background with a rounded corner shape to the navigation bar.
- Modified the secondary color to `0xFFF9B533`.
- Updated the icon of `Feed` from `Outlined.Home` to `Filled.Home`.
- Updated the icon of `Chat` from `Outlined.MailOutline` to `Filled.Email`.
- Updated the icon of `Notification` from `Outlined.Notifications` to `Filled.Notifications`.
- Updated the icon of `Profile` from `Outlined.Person` to `Filled.Person`.
- Added colors to the NavigationBarItem.

FC4RICA committed:
Refactor: Update navigation bar style and icons, and modify secondary color

- Changed the navigation bar's style by setting its container color to primary, content color to onPrimary, and adding rounded corners.
- Added a background with a rounded corner shape to the navigation bar.
- Modified the secondary color to `0xFFF9B533`.
- Updated the icon of `Feed` from `Outlined.Home` to `Filled.Home`.
- Updated the icon of `Chat` from `Outlined.MailOutline` to `Filled.Email`.
- Updated the icon of `Notification` from `Outlined.Notifications` to `Filled.Notifications`.
- Updated the icon of `Profile` from `Outlined.Person` to `Filled.Person`.
- Added colors to the NavigationBarItem.

FC4RICA committed:
Refactor: Update navigation bar style and icons, and modify secondary color

- Changed the navigation bar's style by setting its container color to primary, content color to onPrimary, and adding rounded corners.
- Added a background with a rounded corner shape to the navigation bar.
- Modified the secondary color to `0xFFF9B533`.
- Updated the icon of `Feed` from `Outlined.Home` to `Filled.Home`.
- Updated the icon of `Chat` from `Outlined.MailOutline` to `Filled.Email`.
- Updated the icon of `Notification` from `Outlined.Notifications` to `Filled.Notifications`.
- Updated the icon of `Profile` from `Outlined.Person` to `Filled.Person`.
- Added colors to the NavigationBarItem.

FC4RICA committed:
Refactor: Update navigation bar style and icons, and modify secondary color

- Changed the navigation bar's style by setting its container color to primary, content color to onPrimary, and adding rounded corners.
- Added a background with a rounded corner shape to the navigation bar.
- Modified the secondary color to `0xFFF9B533`.
- Updated the icon of `Feed` from `Outlined.Home` to `Filled.Home`.
- Updated the icon of `Chat` from `Outlined.MailOutline` to `Filled.Email`.
- Updated the icon of `Notification` from `Outlined.Notifications` to `Filled.Notifications`.
- Updated the icon of `Profile` from `Outlined.Person` to `Filled.Person`.
- Added colors to the NavigationBarItem.

Thanachot-W committed:
register

FC4RICA committed:
Refactor: Implement custom typography with Noto Sans Thai font family

- Added Noto Sans Thai font family to project resources.
- Created a `BonBaanTypography` variable to define typography using the Noto Sans Thai font family.
- Added a `defaultFontFamily` extension function to easily apply a font family to all styles within `Typography`.
- Updated `BonBaanTheme` to use `BonBaanTypography` instead of the default `Typography`.

FC4RICA committed:
Refactor: Implement custom typography with Noto Sans Thai font family

- Added Noto Sans Thai font family to project resources.
- Created a `BonBaanTypography` variable to define typography using the Noto Sans Thai font family.
- Added a `defaultFontFamily` extension function to easily apply a font family to all styles within `Typography`.
- Updated `BonBaanTheme` to use `BonBaanTypography` instead of the default `Typography`.

FC4RICA committed:
Refactor: Implement custom typography with Noto Sans Thai font family

- Added Noto Sans Thai font family to project resources.
- Created a `BonBaanTypography` variable to define typography using the Noto Sans Thai font family.
- Added a `defaultFontFamily` extension function to easily apply a font family to all styles within `Typography`.
- Updated `BonBaanTheme` to use `BonBaanTypography` instead of the default `Typography`.

FC4RICA committed:
Refactor: Implement custom typography with Noto Sans Thai font family

- Added Noto Sans Thai font family to project resources.
- Created a `BonBaanTypography` variable to define typography using the Noto Sans Thai font family.
- Added a `defaultFontFamily` extension function to easily apply a font family to all styles within `Typography`.
- Updated `BonBaanTheme` to use `BonBaanTypography` instead of the default `Typography`.

FC4RICA committed:
Refactor: Replace BottomNavBar with BonBaanBottomNavBar and improve appearance

- Replaced `BottomNavBar` with `BonBaanBottomNavBar` for better naming.
- Added a transparent `Surface` with rounded corners for background.
- Used `WindowInsets.navigationBars` to manage insets.
- Added a transparent `containerColor` in `NavigationBar`.
- Moved `HomeSection` into a `Box` to manage background and navbar layers.

FC4RICA committed:
Refactor: Replace BottomNavBar with BonBaanBottomNavBar and improve appearance

- Replaced `BottomNavBar` with `BonBaanBottomNavBar` for better naming.
- Added a transparent `Surface` with rounded corners for background.
- Used `WindowInsets.navigationBars` to manage insets.
- Added a transparent `containerColor` in `NavigationBar`.
- Moved `HomeSection` into a `Box` to manage background and navbar layers.

FC4RICA committed:
Refactor: Replace BottomNavBar with BonBaanBottomNavBar and improve appearance

- Replaced `BottomNavBar` with `BonBaanBottomNavBar` for better naming.
- Added a transparent `Surface` with rounded corners for background.
- Used `WindowInsets.navigationBars` to manage insets.
- Added a transparent `containerColor` in `NavigationBar`.
- Moved `HomeSection` into a `Box` to manage background and navbar layers.

FC4RICA committed:
Refactor: Replace BottomNavBar with BonBaanBottomNavBar and improve appearance

- Replaced `BottomNavBar` with `BonBaanBottomNavBar` for better naming.
- Added a transparent `Surface` with rounded corners for background.
- Used `WindowInsets.navigationBars` to manage insets.
- Added a transparent `containerColor` in `NavigationBar`.
- Moved `HomeSection` into a `Box` to manage background and navbar layers.

FC4RICA committed:
Merge branch 'feature/typography' of https://github.com/FC4RICA/BonBaan into feature/typography

FC4RICA committed:
Refactor: Introduce BonBaanButton and update RegisterScreen

- Added `BonBaanButton` composable for customizable button styles.
- Renamed and moved `ButtonComponent` to `BonBaanButton` and `component` to `com.fc4rica.bonbaan.ui.components`.
- Implemented various `ButtonVariant` styles: PRIMARY, SECONDARY, OUTLINED, TEXT, and DESTRUCTIVE.
- Updated `RegisterScreen` to use `BonBaanButton`.
- Added `Preview` for `BonBaanButton` and `RegisterScreen`.
- Corrected the package of `Button` and `TextField` component.
- Corrected start destination of authenticationGraph.

FC4RICA committed:
Refactor: Update RegisterScreen UI and add BonBaanTextField

- Added `BonBaanTextField` and updated `RegisterScreen` to utilize it.
- Updated `Button` component, adding `TEXT` variant.
- Updated `RegisterScreen` layout and content, with improved text elements.
- Replaced `TextFieldComponent` with `BonBaanTextField` in `RegisterScreen`.
- Added password visibility toggle to `BonBaanTextField`.
- Updated navigation in `RegisterScreen` to navigate to `LoginScreen`.
- Added `ContentPadding` to `BonBaanButton`.

FC4RICA committed:
Feat: Add logo to RegisterScreen and update UI

- Added an app logo to the `RegisterScreen` using the `Image` composable and the `logo1` drawable resource.
- Updated the UI of the `RegisterScreen` to include the logo, adjust spacing, and correct spelling:
  - Added an `Image` composable to display the `logo1` with specific dimensions and padding.
  - Introduced `Spacer` composables for vertical spacing between elements.
  - Corrected the spelling of "อีเมล์" to "อีเมล" in the `BonBaanTextField` label and the body text.
- Added logo1.png file.

FC4RICA committed:
Refactor: Update RegisterScreen navigation and button actions

- Removed navigation from the "Next" button in `RegisterScreen`.
- Added navigation from the "Login" button in `RegisterScreen` to the `Screen.Login` route.

guayJheng committed:
Feat: Add homepage

guayJheng committed:
Feat: SearchScreen

guayJheng committed:
Feat: SearchScreen

guayJheng committed:
Refactor: Update RecommendItem

guayJheng committed:
Refactor: Update RecommendItem

FC4RICA committed:
Refactor: Remove ChatScreen and PreviousVow screens and introduce VowRecord screen

- Removed `ChatScreen` and related `Screen.Chat` navigation.
- Removed `PreviousVowDetailScreen` and `PreviousVowsScreen` with their related navigation.
- Introduced `VowRecordScreen` and `Screen.VowRecord` for recording vows.
- Changed "home_chat" string to "home_vow_record".
- Replaced `Icons.Filled.Email` with `Icons.AutoMirrored.Filled.StickyNote2` for VowRecord.
- Updated `HomeSection.sections` to reflect the removal of Chat and addition of VowRecord.

FC4RICA committed:
Merge branch 'fix/move-prev-vow-screen' of https://github.com/FC4RICA/BonBaan into fix/move-prev-vow-screen
fix: merge conflict

FC4RICA committed:
Refactor: Update package and rename VowRecordScreen

- Moved `VowRecordScreen` to the `ui.home` package.
- Removed the redundant `SearchScreen` function.

FC4RICA committed:
Merge branch 'feature/bottom-navigation-bar' of https://github.com/FC4RICA/BonBaan into feature/bottom-navigation-bar

FC4RICA committed:
Merge branch 'feature/home-screen-navigation' of https://github.com/FC4RICA/BonBaan into feature/home-screen-navigation

FC4RICA committed:
Merge branch 'feature/typography' of https://github.com/FC4RICA/BonBaan into feature/typography

FC4RICA committed:
Refactor: Restructure register flow and rename RegisterScreen

- Renamed `RegisterScreen` to `InputEmailScreen`.
- Created new composable screens: `EmailVerificationScreen`, `PersonalInfoScreen`, and `PasswordSetupScreen`.
- Added `registerGraph` to handle the navigation flow for the registration process.
- Updated navigation to include the new register flow with `InputEmail`, `EmailVerification`, `PersonalInfo`, and `PasswordSetup`.
- Modified `Auth.kt` to use `registerGraph`.
- Added new routes in `Screen.kt` for `InputEmail`, `EmailVerification`, `PersonalInfo`, and `PasswordSetup`.

guayJheng committed:
Refactor : Add Icon,BonBaan Component to Homepage

guayJheng committed:
Refactor: Add Icons,BonBann Component to SearchScreen

guayJheng committed:
Refactor: Add Icons,BonBann Component to SearchScreen

FC4RICA committed:
Feat: Implement Auth Screens

- Added `rememberImeState` to manage keyboard visibility for dynamic UI adjustments in `InputEmailScreen`.
- Implemented dynamic UI adjustment based on keyboard visibility in `InputEmailScreen`, adjusting the logo size accordingly.
- Added `PasswordSetupScreen` for user password creation and confirmation.
- Created `EmailVerificationScreen` for email verification via OTP input and added resend option.
- Implemented `PersonalInfoScreen` to allow user input of name, phone number, and username.
- Added `TextAlign.Center` in each screen to center the helper text.
- Improved spacing and padding in each screen.
- Added `isEnabled` for "Next" button in `EmailVerificationScreen` to validate OTP length.
- Added `isPassword` in `BonBaanTextField` to input password.

FC4RICA committed:
Feat: Implement navigation and UI enhancements for authentication flow

- Added navigation transitions between authentication screens: `InputEmail`, `EmailVerification`, `PersonalInfo`, and `PasswordSetup`.
- Implemented `OtpInputField` for entering verification codes.
- Added a logo to the `PasswordSetup`, `EmailVerification`, and `PersonalInfo` screens.
- Updated `InputEmailScreen` and `EmailVerificationScreen` to navigate to the next screen on button click.
- Implemented `PasswordSetupScreen` to navigate to the `Onboarding` screen.
- Implemented `PersonalInfoScreen` to navigate to the `PasswordSetup` screen.
- Added `logo2.png` as an app logo.

Thanachot-W committed:
interest

FC4RICA committed:
Feat: Implement Splash Screen and update dependencies

- Implemented the Android Splash Screen using the `SplashScreen` API and compat library.
- Installed the splash screen in `MainActivity` using `installSplashScreen()`.
- Created a `Theme.BonBaanSplashScreen` style to customize the splash screen, including background color and icon.
- Updated `AndroidManifest.xml` to use the new splash screen theme.
- Added `core-splashscreen` dependency and updated the `libs.versions.toml` file with `coreSplashscreenVersion`
- Updated  Compose BOM version to "2025.03.00" and navigation libraries version.
- Added `purple` color in `colors.xml`.
- Added the `splash_logo` as splash screen icon
- Removed unused colors in `colors.xml`

FC4RICA committed:
Merge pull request #2 from FC4RICA/feature/home-screen-navigation

feature/home-screen-navigation

FC4RICA committed:
Merge branch 'development' into feature/bottom-navigation-bar

FC4RICA committed:
Merge pull request #5 from FC4RICA/feature/bottom-navigation-bar

feature/bottom-navigation-bar

FC4RICA committed:
Merge branch 'development' into feature/typography

FC4RICA committed:
Merge pull request #6 from FC4RICA/feature/typography

feature/typography

FC4RICA committed:
Merge pull request #7 from FC4RICA/fix/move-prev-vow-screen

fix/move-prev-vow-screen

FC4RICA committed:
Merge pull request #9 from FC4RICA/feature/register-pages

feature/register-pages

FC4RICA committed:
Merge pull request #10 from FC4RICA/feature/splash-screen

feature/splash-screen

FC4RICA committed:
Refactor: Implement user registration flow and related components

- Added `ValidationUtils` for email validation.
- Implemented `UserRepositoryImpl` with dummy data for user operations.
- Created `OtpUseCase`, `UserRepository`, `UserUseCase`, `LoginUseCase`, and `RegisterUseCase` for handling user-related logic.
- Created `RegisterUiState` and `RegisterViewModel` to manage the state of the registration process.
- Added `registerGraph` for navigating the user registration flow.
- Created `InputEmailScreen` to collect the user's email address.
- Added `User`, `RegisterRequest` data models.
- Updated `authenticationGraph` to include the registration flow and `RegisterViewModel`.
- Added `viewModel` dependency.
- Added `viewmodel-compose` dependency.

FC4RICA committed:
Refactor: Inject UserUseCase into BonBaanNavHost and BonBaanApp

- Introduced `UserUseCase` as a parameter in `BonBaanNavHost`.
- Passed `UserUseCase` from `BonBaanApp` to `BonBaanNavHost`.
- Removed redundant `UserRepository` instantiation from `BonBaanNavHost`.

FC4RICA reviewed (changes requested):
Don't forget to use Theme Typography and ColorScheme

Thanachot-W committed:
feature/interest

-use MaterialTheme.colorScheme
-use MaterialTheme.typography

FC4RICA committed:
Refactor: Integrate Koin for dependency injection and update navigation

- Implemented Koin for dependency injection:
    - Created `AppModule` to define dependencies and `initKoin` to initialize Koin.
    - Moved `RegisterViewModel` instantiation to Koin.
    - Injected `RegisterViewModel` into `InputEmailScreen` using `koinViewModel`.
    - Replaced manual dependency injection with Koin in `BonBaanApp`.
    - Removed `provideFactory` from `RegisterViewModel`.
- Updated navigation:
    - Removed `registerViewModel` parameter from `authenticationGraph` and `registerGraph` functions.
    - Removed `userUseCase` parameter from `BonBaanNavHost`.
- Added Koin dependencies to `libs.versions.toml`.
- Updated dependencies in `build.gradle.kts`.

FC4RICA committed:
Refactor: Remove UseCase classes and update RegisterViewModel

- Removed `LoginUseCase`, `OtpUseCase`, `RegisterUseCase`, and `UserUseCase` classes.
- Updated `RegisterViewModel` to directly use `UserRepository`.
- Modified `User` model to include `firstname` and `lastname` instead of `name`.
- Renamed `otp` function in `UserRepository` to `sendOtp`.
- Updated `RegisterViewModel` to send and process data to the repository.
- Removed `UserUseCase` from the dependency injection.
- Update to send to repository in RegisterViewModel.

FC4RICA committed:
Refactor: Enhance user registration flow and input handling

- Modified `InputEmailScreen` to navigate to `PersonalInfo` screen instead of `EmailVerification`.
- Changed `InputEmailScreen` to use `updateField` method from view model.
- Modified `PasswordSetupScreen` to navigate to `EmailVerification` screen instead of `Onboarding`.
- Changed `PasswordSetupScreen` to use view model for password and confirm password.
- Introduced `updateField` function to `RegisterViewModel` for managing various input fields.
- Changed `EmailVerificationScreen` to use view model for storing the code and code validation.
- Added resend code action to `EmailVerificationScreen` button.
- Modified `PersonalInfoScreen` to use view model for storing data.
- Added `isCodeValid` to the view model's state.

FC4RICA committed:
Refactor: Enhance email input validation and update UI

- Added email validation logic to `RegisterViewModel` using the `isValidEmail` utility function.
- Updated the `InputEmailScreen` to display an error message when the entered email is invalid.
- Modified the "Next" button in `InputEmailScreen` to navigate only if the entered email is valid.
- Added preview capability to `InputEmailScreen` using a preview module.
- Modified `RegisterViewModel` to update email validity and introduced submit email function.

FC4RICA committed:
Feat: Implement input validation for email, name, phone, and username

- Added input validation for email format in `InputEmailScreen`.
- Added validation utilities for name, phone, and username in `ValidationUtils.kt`.
- Added validation logic for email, name, phone, and username in `RegisterViewModel`.
- Implemented error message display for invalid email, name, phone, and username in `InputEmailScreen` and `PersonalInfoScreen`.
- Updated the `PersonalInfoScreen` to navigate to the next screen only if the name, phone, and username are valid.
- Updated `RegisterViewModel` to support validating name, phone, and username.
- Updated `RegisterViewModel` to only accept numeric characters in the phone number field.

FC4RICA committed:
Feat: Implement password validation and OTP resend logic

- Added password validation logic to `PasswordSetupScreen` using `isValidPassword` and `isValidConfirmPassword`.
- Added error messages to `PasswordSetupScreen` to provide feedback on invalid input.
- Implemented resending OTP logic with a cooldown timer in `EmailVerificationScreen` using `otpCooldown`.
- Added logic to `RegisterViewModel` to manage password validation and OTP sending, including handling OTP cooldown, sending, and resending.
- Updated `RegisterUiState` to include `passwordError`, `confirmPasswordError`, and `isOtpSent` for UI feedback and logic control.
- Added `onEnterEmailVerificationScreen` to send OTP once on screen load.
- Added preview dependencies to fix the koin issue.

FC4RICA committed:
Refactor: Restructure register process and view models

- Removed `RegisterViewModel` and introduced separate view models for each registration step: `InputEmailViewModel`, `PersonalInfoViewModel`, `PasswordSetupViewModel`, and `EmailVerificationViewModel`.
- Created `RegisterRepository` and `RegisterRepositoryImpl` to manage the registration data flow and state, including the `RegisterRequest`.
- Updated `AppModule` to use the new view models and repository.
- Implemented validation within the new view models.
- Moved the `RegisterRequest` to be managed by `RegisterRepository` and accessible through the view models.
- Moved `UserRepository` to `RegisterRepository`
- Updated the UI components (`InputEmailScreen`, `PersonalInfoScreen`, `PasswordSetupScreen`, `EmailVerificationScreen`) to use the new view models and data flow.
- Removed unused code and dependencies.

FC4RICA committed:
Refactor: Implement navigation and update register flow

-   Refactored `InputEmailScreen`, `PasswordSetupScreen`, `PersonalInfoScreen`, `EmailVerificationScreen`, and `Register` to manage navigation using callbacks instead of directly accessing `NavHostController`.
-   Introduced `navigateToLogin`, `navigateToPersonalInfo`, `navigateToSetupPassword` , `navigateToEmailVerification` and `navigateToOnboarding` lambdas for navigating between screens.
-   Implemented custom slide transitions when navigating within the registration flow using `AnimatedContentTransitionScope.SlideDirection`.
-   Populated name in `PersonalInfoScreen` from `registerRequest` in `PersonalInfoViewModel` if available.

FC4RICA committed:
Merge pull request #14 from FC4RICA/feature/view-model

feature/view-model

Thanachot-W committed:
feature/interest

- move FilterChip to Component Folder
- merge Interestelection and Onboardingheader into InterestScreen

FC4RICA committed:
Merge pull request #8 from FC4RICA/feature/interest

interest

guayJheng committed:
Refactor Update Theme Typography and ColorScheme

guayJheng committed:
Refactor Update Theme Typography and ColorScheme

guayJheng committed:
Refactor Update Theme Typography and ColorScheme

guayJheng committed:
Refactor Typography

guayJheng committed:
Merge branch 'feature/homepage-recommend' into development

guayJheng committed:
Merge branch 'feature/home-searchScreen' of https://github.com/FC4RICA/BonBaan into feature/home-searchScreen
